R/GoogleData.R

```diff
@@ -0,0 +1,170 @@
+        full_join(
+          self$data$raw$hospitalizations,
+          by = c("key", "date")
+        ) %>%
+        select(
+          .data$date,
+          .data$`3166-1-alpha-3`,
+          .data$country_name,
+          .data$subregion1_code,
+          .data$subregion1_name,
+          .data$new_confirmed,
+          .data$total_confirmed,
+          .data$new_deceased,
+          .data$total_deceased,
+          .data$new_recovered,
+          .data$total_recovered,
+          .data$new_tested,
+          .data$total_tested,
+          .data$new_hospitalized,
+          .data$total_hospitalized
+        ) %>%
+        mutate(
+          date = ymd(.data$date),
+          new_confirmed = as.numeric(.data$new_confirmed),
+          total_confirmed = as.numeric(.data$total_confirmed),
+          new_deaths = as.numeric(.data$new_deceased),
+          total_deaths = as.numeric(.data$total_deceased),
+          new_recovered = as.numeric(.data$new_recovered),
+          total_recovered = as.numeric(.data$total_recovered),
+          new_tested = as.numeric(.data$new_tested),
+          total_tested = as.numeric(.data$total_tested),
+          new_hospitalized = as.numeric(.data$new_hospitalized),
+          total_hospitalized = as.numeric(.data$total_hospitalized)
+        ) %>%
+        rename(
+          level_1_region = .data$country_name,
+          level_1_region_code = .data$`3166-1-alpha-3`,
+          level_2_region_code = .data$subregion1_code,
+          level_2_region = .data$subregion1_name,
+          cases_new = .data$new_confirmed,
+          cases_total = .data$total_confirmed,
+          deaths_new = .data$new_deaths,
+          deaths_total = .data$total_deaths,
+          tested_new = .data$new_tested,
+          tested_total = .data$total_tested,
+          recovered_new = .data$new_recovered,
+          recovered_total = .data$total_recovered,
+          hosp_new = .data$new_hospitalized,
+          hosp_total = .data$total_hospitalized
+        ) %>%
+        replace_na(
+          list(
+            level_1_region = "Unknown",
+            level_2_region = "Unknown"
+          )
+        )
+    },
+
+    #' @description JHU specific country level data cleaning
+    #' @importFrom dplyr select summarise group_by
+    #' @importFrom rlang .data
+    clean_level_1 = function() {
+      self$data$clean <- self$data$clean %>%
+        select(
+          .data$date,
+          .data$level_1_region_code, .data$level_1_region,
+          .data$cases_new, .data$cases_total,
+          .data$deaths_new, .data$deaths_total,
+          .data$tested_new, .data$tested_total,
+          .data$recovered_new, .data$recovered_total,
+          .data$hosp_new, .data$hosp_total
+        ) %>%
+        group_by(
+          .data$date, .data$level_1_region_code,
+          .data$level_1_region
+        ) %>%
+        summarise(across(where(is.double), sum))
+    },
+
+    #' @description Specific return settings for the JHU dataset.
```

Comment:
Is this copy and pasted across JHU and google? Do we really need this it is quite clunky?

Reply:
Nah, will remove